collectors_app.php

```diff
@@ -21,68 +21,108 @@
 
 // html form ✅
 // $_GET form data ✅
-// move data to DB
+// move data to DB ✅
 
-//----------------------------------------------------------------------------------------------------------------------empty socks array
-$socks = [];
+// trainer tasks:
 
-//----------------------------------------------------------------------------------------------------------------------connect to DB
-$db = new PDO(
-    'mysql:host=DB;dbname=collection',
-    'root',
-    'password'
-);
-//----------------------------------------------------------------------------------------------------------------------defaulting
-$db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
-//----------------------------------------------------------------------------------------------------------------------preparing query
-$query = $db->prepare('SELECT * FROM `socks`');
-//----------------------------------------------------------------------------------------------------------------------execute query
-$result = $query->execute();
+// validation of inputs:
+// DB table for sizes, color, pattern ✅
+// dropdowns that populate colors, sizes, patterns ✅
 
-if ($result) {
-    $socks = $query->fetchAll();//--------------------------------------------------------------------------------------REASSIGN PLAYERS ARRAY
-} else {
-    echo 'not working';
+//----------------------------------------------------------------------------------------------------------------------dropdowns
+//----------------------------------------------------------------------------------------------------------------------connect DB function
+function connectDB() {
+    $db = new PDO(
+        'mysql:host=DB;dbname=collection',
+        'root',
+        'password'
+    );
+    $db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+    return $db;
 }
-//----------------------------------------------------------------------------------------------------------------------form $_GET data
-$test = '';
-if ($_GET) {
-    $test = "{$_GET['size']} {$_GET['pattern']} {$_GET['color']}";
-
-    //------------------------------------------------------------------------------------------------------------------move form to DB
-    $query = $db->prepare("INSERT INTO `socks` (`size`, `pattern`, `color`) VALUES (:size, :pattern, :color)");
-    $result = $query->execute([
-        'size' => $_GET['size'],
-        'pattern' => $_GET['pattern'],
-        'color' => $_GET['color']
-    ]);
+//----------------------------------------------------------------------------------------------------------------------get table function
+function getTable($table, $db) {
+    $query = $db->prepare("SELECT `{$table}` FROM `{$table}s`");
+    $result = $query->execute();
+    if ($result) {
+        return $query->fetchAll();
+    } else {
+        echo 'not working'; //maybe return not working
```

Comment:
Yeah this will crash if it doesn't return anything. 

Maybe throw an exception instead of echo